src/game.cpp

```diff
@@ -8516,39 +8536,45 @@ void game::zones_manager()
             } else if( action == "CONFIRM" ) {
                 auto &zone = zones[active_index].get();
 
-                uimenu as_m;
+                uilist as_m;
                 as_m.text = _( "What do you want to change:" );
                 as_m.entries.emplace_back( uimenu_entry( 1, true, '1', _( "Edit name" ) ) );
                 as_m.entries.emplace_back( uimenu_entry( 2, true, '2', _( "Edit type" ) ) );
                 as_m.entries.emplace_back( uimenu_entry( 3, zone.get_options().has_options(), '3',
                                            _( "Edit options" ) ) );
                 as_m.entries.emplace_back( uimenu_entry( 4, true, '4', _( "Edit position" ) ) );
-                as_m.entries.emplace_back( uimenu_entry( 5, true, 'q', _( "Cancel" ) ) );
                 as_m.query();
 
                 switch( as_m.ret ) {
                     case 1:
-                        zone.set_name();
-                        stuff_changed = true;
+                        if( zone.set_name() ) {
+                            stuff_changed = true;
```

Comment:
what about
`stuff_changed = zone.set_name();`

Reply:
That will break if something changes `stuff_changed` before. I could of course write it as `stuff_changed |= zone.set_name()`, but it might be clearer this way.

Reply:
how about 

stuff_changed = stuff_changed || zone.set_name();

that's clear enough and not spreaded across 3 lines

Reply:
nah, leave it as it is, it's minor detail and personal preference more than anything else